mitmproxy/tools/console/flowview.py

```diff
@@ -115,26 +115,21 @@ def view_tcp_stream(self) -> urwid.Widget:
         if not flow.messages:
             return searchable.Searchable([urwid.Text(("highlight", "No messages."))])
 
-        from_client = None
-        messages = []
-        for message in flow.messages:
-            if message.from_client is not from_client:
-                messages.append(message.content)
-                from_client = message.from_client
-            else:
-                messages[-1] += message.content
+        viewmode = self.master.commands.call("console.flowview.mode")
 
-        from_client = flow.messages[0].from_client
         parts = []
-        for message in messages:
-            parts.append(
-                (
-                    "head" if from_client else "key",
-                    message
-                )
-            )
-            from_client = not from_client
-        return searchable.Searchable([urwid.Text(parts)])
+        for message in flow.messages:
+            _, lines, _ = contentviews.get_tcp_content_view(viewmode, message.content)
+
+            for line in lines:
+                if message.from_client:
+                    line.insert(0, "--> ")
```

Comment:
How about we use something like `⇒` and ` ⇐`?

We can integrate this here:
https://github.com/mitmproxy/mitmproxy/blob/2774928319b706d8b6f85919f8811776342986c9/mitmproxy/tools/console/common.py#L100-L113

Also, it would be super awesome if we can color-code the arrows as blue and red. That helps a bit to determine message boundaries 😃 #3929 shows how you can define palette entries. I very hackily used "head" and "key" before.

Reply:
This will make arrows more compact which will free space for an actual messages! Totally agree, thanks!